imagebuildah/stage_executor.go

```diff
@@ -1175,6 +1199,13 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 		// image, run the step so that we can check if the result
 		// matches a cache.
 		if cacheID == "" {
+			// Note: In theory we should never ever hit this branch now
+			// since theoretically only `COPY` and `ADD` should  fall in
```

Comment:
RUN does not adds any content so I don't know how digest needs to updated. But let me verify this. If there is a `cache` hit for RUN it mostly hits in the first `PASS` althought but `ARG` for sure needs to get evaluated so it might hit this branch.

Reply:
`RUN dnf install httpd`, if it doesn’t happen to match a cache, absolutely must end up an `ib.Run` call, if I understand what `ib.Run` does correctly.

(And it also definitely adds content, although we don’t compute a digest of that and we don’t seem to restrict cache matches to exact match of the added content.)

Reply:
@mtrmac Yes my bad i meant this one: https://github.com/containers/buildah/pull/4128#discussion_r926777116

Reply:
Should we discuss here to avoid duplication: https://github.com/containers/buildah/pull/4128#discussion_r926699466 so resolving this thread. But we can re-open if this is different than this https://github.com/containers/buildah/pull/4128#discussion_r926699466